common/remote/sockfile.cpp

```diff
@@ -4012,7 +4012,15 @@ class CRemoteDiskReadActivity : public CRemoteDiskBaseActivity
             cursorDirty = false;
             return;
         }
+
         OwnedIFile iFile = createIFile(fileName);
+
+        // NB: auto detect if file is compressed or not.
+        bool autoCompressed = isCompressedFile(iFile);
```

Comment:
What is the danger of false positives?  In nearly all other places it is only checked if it is thought to be compressed.

You could call createCompressedFileReader without this check, which I think returns null if it is not a compressed file.  Avoids reading the end of the file twice.

Reply:
No real danger of false positives and I did consider taking out the meta flag altogether.
But decided to keep and warn for info. only if meta flagged compressed and it wasn't. 

Yes createCompressedFileReader could be called directly as it calls isCompressedFile itself, so I'll make that change.